card_live_dashboard/layouts/figures.py

```diff
@@ -114,6 +122,210 @@ def totals_figure(data: CardLiveData, type_value: str, color_by_value: str) -> g
     return fig
 
 
+def rgi_intersection_figure(data: CardLiveData, type_value: str) -> go.Figure:
+    """
+    Generates an upset plot figure in plotly based on the presence/absence of
+    the selected category across filenames in the RGI output selected
+    :param data: a CardLiveData object from which the rgi_parser is called
+    :param type_value: The category in RGI to plot set membersips for
+    :return: A plotly figure object for displaying via dash, returned figure
+             is empty or gives a warning requesting further subsetting if
+             there is not enough or too much data to meaningfully generate
+             subsets
+    """
+
+    if data.empty:
+        fig = EMPTY_FIGURE
+    else:
+        # get category set memberships from rgi data
+        title = RGI_TITLES[type_value]
+        totals_df = data.rgi_parser.get_column_values(data_type=type_value,
+                                                      values_name='categories',
+                                                      drop_duplicates=True)
+        totals_df = totals_df.dropna()
+        category_sets = totals_df.reset_index().groupby('filename')\
+                                        .agg(lambda x: tuple(x)).applymap(list)
+        category_sets = category_sets['categories']\
+                         .apply(lambda x: sorted(x)).sort_values().apply(tuple)
+        category_sets = category_sets.value_counts()
+
+        # convert to upset data
+        upset_data = upsetplot.from_memberships(category_sets.index,
+                                                category_sets.values)
+        upset_data = upsetplot.UpSet(upset_data,
+                                     sort_by='cardinality')
+
+        # get number of sets and categories into UpSet class
+        num_sets, num_categories = upset_data.intersections.reset_index().shape
+
+        # if the data is empty
+        if num_sets == 0 or num_categories == 0:
+            fig = EMPTY_FIGURE
+        # or if there is way too much data to plot
+        elif num_categories > 40:
+            fig = go.Figure(layout={
+                                'xaxis': {'visible': False},
+                                'yaxis': {'visible': False},
+                                'annotations': [{
+                                'text': f"{num_categories} total {title} "
+                                         "categories with current selection. "
+                                         "<br>Please subset input data "
+                                         "further to display set "
+                                         "intersections.",
+                                'xref': 'paper',
+                                'yref': 'paper',
+                                'showarrow': False,
+                                'font': {'size': 28}
```

Comment:
The font size here looks slightly too large when I ran:
![image](https://user-images.githubusercontent.com/200517/102529665-d8e9af00-4065-11eb-9e64-231e47784acd.png)

Reply:
Made smaller.